app/models/container_deployment.rb

```diff
@@ -0,0 +1,11 @@
+class ContainerDeployment < ApplicationRecord
+  has_many :container_deployment_authentications
+  belongs_to :deployed_ems, :class_name => 'ExtManagementSystem'
+  belongs_to :deployed_on_ems, :class_name => 'ExtManagementSystem'
+  belongs_to :automation_task
+  has_many :container_deployment_nodes
+  has_many :container_deployment_custom_setting
+  has_many :container_volumes, :foreign_key => :parent_id
+  serialize :customize, Hash
+  DEPLOYMENT_TYPES = ['origin', 'enterprise'].freeze
```

Comment:
I dont know what this will be used for. If this is some sort of switching on subclasses, I'd rather go for registering something from the subclass.